test/app-luatest/init_script_test.lua

```diff
@@ -0,0 +1,144 @@
+-- The test tests init script.
+
+local t = require('luatest')
+local treegen = require('luatest.treegen')
+local justrun = require('luatest.justrun')
+
+local g = t.group()
+
+-- Access to box.cfg from init script.
+g.test_access_box_cfg_init_script= function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_access_box_cfg_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+box.cfg({memtx_memory = 107374182})
+print('box.cfg.memtx_memory ' .. tostring(box.cfg.memtx_memory))
+os.exit()
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.stdout, 'box.cfg.memtx_memory 107374182')
+end
+
+-- The testcases tests insert tuples using fiber in init script.
+g.test_insert_tuples_fiber_init_script = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_insert_tuples_fiber_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+local fiber = require('fiber')
+local json = require('json')
+
+box.cfg()
+local space = box.schema.space.create('tweedledum')
+space:create_index('primary', { type = 'hash' })
+
+function do_insert()
+    space:insert({1, 2, 4, 8})
+end
+
+fiber.create(do_insert)
+
+print(json.encode(space:select(nil, { limit = 4 })))
+
+os.exit()
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.stdout, '[[1,2,4,8]]')
+    t.assert_equals(res.exit_code, 0)
+end
+
+-- The testcase tests insert from detached fiber.
+g.test_insert_tuples_and_select_init_script = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_insert_tuples_and_select_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+local yaml = require("yaml")
+local fiber = require("fiber")
+
+box.cfg()
+local space = box.schema.space.create("tweedledum")
+space:create_index("primary", { type = "hash" })
+
+fiber.create(function()
+    space:insert({1, 2, 4, 8})
+end)
+
+print(yaml.encode(space:select()))
+
+os.exit(0)
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.stdout, [[
+---
+- [1, 2, 4, 8]
+...]])
+end
+
+-- The testcase tests insert from an init script.
+g.test_insert_tuples_init_script = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_insert_tuples_and_get_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+box.cfg()
+local space = box.schema.space.create("tweedledum")
+space:create_index("primary", { type = "hash" })
+
+space:insert({2, 4, 8, 16})
+print(space:get(1))
+print(space:get(2))
+print(space:get(3))
+print(space:get(4))
```

Comment:
Nit: with `space:get(<...>) or 'nil'` it would be more explicit.

Reply:
Did you mean the same and without `print`?

Reply:
No-no, I meant that the first, the third and the fourth statements are producing just a newline and it is not very obvious from the code (especially, because the resulting output is rstripped). I propose to do `print(space:get(<N>) or 'nil')` to make the output more clear.

Reply:
Fixed:

```diff
 space:insert({2, 4, 8, 16})
-print(space:get(1))
-print(space:get(2))
-print(space:get(3))
-print(space:get(4))
+print(space:get(1) or 'nil')
+print(space:get(2) or 'nil')
+print(space:get(3) or 'nil')
+print(space:get(4) or 'nil')
 
 os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0, 'exit code')
-    t.assert_equals(res.stdout, '\
-[2, 4, 8, 16]', 'stdout')
+    t.assert_equals(res.stdout, 'nil\
+[2, 4, 8, 16]\
+nil\
+nil', 'stdout')
 end
 
 -- The testcase checks that `math.floor` is reachable in the init
```